### PR TITLE
feat: allow row height setting in glide table

### PR DESCRIPTION
**Pull request by ashtonG**
## Description
This adds a new table settings menu that will hold options that don't change the result set but do change how the table looks. The first option we're adding is the "row height" setting, which changes the table information density.

## Test Plan

* options menu appears on glide table
* clicking a row height option on the glide table affects the row height on the table
* row height preference is preserved on reload

## Commentary (optional)

@alexjdetermined some feedback needed:
* Are the row heights for comfy and compact appropriate?
  *  comfy (48px):
     image
  
  * compact (32px):
    image

* I think the wireframes had this menu behind an icon -- is there an svg for that somewhere?
* The compact menu icon looks uneven on high dpi displays -- do we have an alternate option for this?:
  image

## Checklist

- [x] Changes have been manually QA'd
- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket
WEB-994

### CODE REVIEW

**Comment by ashtonG** [comment]
The current default height is 40px. Do we want to change that?

File: webui/react/src/pages/F_ExpList/glide-table/icons.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
+export const CompactHeight: React.FC = () => (
```
Comment (johnkim-det) [inline webui/react/src/pages/F_ExpList/glide-table/icons.tsx:1]: should we prefer this to the practice of using `.svg` files in `assets/images`?

File: webui/react/src/pages/F_ExpList/F_ExperimentList.settings.ts [hunk 2]
```diff
@@ -5,9 +5,24 @@ import { SettingsConfig } from 'hooks/useSettings';
 
 import { defaultExperimentColumns } from './glide-table/columns';
 
+export const RowHeight = {
+  DEFAULT: 'DEFAULT',
+  EXTRATALL: 'EXTRATALL',
```
Comment (johnkim-det) [inline webui/react/src/pages/F_ExpList/F_ExperimentList.settings.ts:10]: super nit: snake case?

File: webui/react/src/components/kit/Dropdown.tsx [hunk 3]
```diff
@@ -36,11 +42,15 @@ interface BaseProps {
...
   content?: React.ReactNode;
   menu?: never;
+  selectable?: never;
+  selectedKeys?: never;
 };
 
 type MenuProps = {
   content?: never;
   menu?: MenuItem[];
+  selectable?: boolean;
```
Comment (johnkim-det) [inline webui/react/src/components/kit/Dropdown.tsx:52]: Is there a reason to use a `Dropdown` with `selectable` instead of a `Select`?

**Review by johnkim-det — COMMENTED** [review]
Not sure if it works for "Options" be a dropdown that contains an Option Group for "Row height" -- I think that even if additional option groups are added later, you can only have one item selected within a dropdown at a time anyway.

If we want to prepare for future options to be added later, perhaps "Options" could be a `Tooltip` like the other Action Bar items, with a "Row height" `Select` in it?

Or for now it could just be a "Row height" `Select`, without an "Options" menu.

**Comment by ashtonG** [comment]
@johnkim-det i'm basing this off of the designs in the figma:
https://www.figma.com/file/fI9wKqYd3HLflVCBjb3xOS/experiment-list?type=design&node-id=276-7360&t=8MgerGX4cy7SwOcs-0

Note that the `Menu` component can support marking multiple items as selected but the design only requires the rowheight option to be marked as active. We don't want to use a `Select` for this because the items in the menu should be able to support interactions beyond a simple click.

File: webui/react/src/pages/F_ExpList/glide-table/icons.tsx (hunk 1 above)
Comment (ashtonG) [reply webui/react/src/pages/F_ExpList/glide-table/icons.tsx:1]: I don't have major feelings either way, but `import`ing the files returns a path to the file to be used as a src, which doesn't allow the svg to use the `currentcolor` keyword.

File: webui/react/src/pages/F_ExpList/F_ExperimentList.settings.ts [hunk 4]
```diff
@@ -5,9 +5,24 @@ import { SettingsConfig } from 'hooks/useSettings';
...
+  SHORT: 'SHORT',
+  TALL: 'TALL',
+} as const;
+const ioRowHeight = t.keyof({
```
Comment (ClaireNeveu) [inline webui/react/src/pages/F_ExpList/F_ExperimentList.settings.ts:14]: Curious why `t.keyof(RowHeight)` doesn't work.

Comment (ClaireNeveu) [reply webui/react/src/pages/F_ExpList/F_ExperimentList.settings.ts:14]: Also I guess technically it should be valueof.

**Comment by alexjdetermined** [comment]
Looking at the PR the current px heights feel right so no change to 40 for default. 

Re: the multi-option menus. Let's break them out into individual menus component wise and will help with discoverability. 

Design changes: 
* Make button icon only (icon sent via slack)
* Add tooltip on hover (top) that says "Row height"
* add second "button group" to action bar that is aligned right (we can add more of the upcoming "options" as additional icon only buttons to this group as we grow)
* menu items will not have an icon (i dont think it adds much to making it easier to select in this case)

image

File: webui/react/src/pages/F_ExpList/F_ExperimentList.settings.ts (hunk 4 above)
Comment (ashtonG) [reply webui/react/src/pages/F_ExpList/F_ExperimentList.settings.ts:14]: `keyof(RowHeight)` should work here -- just making it apparent for someone looking at the code we're using the values of the enum object, not the keys.

**Comment by johnkim-det** [comment]
> @johnkim-det i'm basing this off of the designs in the figma: https://www.figma.com/file/fI9wKqYd3HLflVCBjb3xOS/experiment-list?type=design&node-id=276-7360&t=8MgerGX4cy7SwOcs-0
> 
> Note that the `Menu` component can support marking multiple items as selected but the design only requires the rowheight option to be marked as active. We don't want to use a `Select` for this because the items in the menu should be able to support interactions beyond a simple click.

Thanks, makes sense seeing the set of dropdowns in the figma. Would be nice to have "selectable" and Option Group examples added to the DesignKit page.

File: webui/react/src/pages/F_ExpList/F_ExperimentList.settings.ts (hunk 4 above)
Comment (ashtonG) [reply webui/react/src/pages/F_ExpList/F_ExperimentList.settings.ts]: added

Comment (ClaireNeveu) [reply webui/react/src/pages/F_ExpList/F_ExperimentList.settings.ts]: Thanks

**Review by alexjdetermined — APPROVED** [review]
LGTM!

**Review by johnkim-det — APPROVED** [review]
still would prefer Dropdown examples added to DesignKit, but otherwise LGTM

**Pull request merged**